content/languages/javascript/authoring.md

```diff
@@ -0,0 +1,281 @@
+---
+kind: tutorial
+languages: [javascript]
+sidebar: "language:javascript"
+prev: /languages/javascript/codewars-test/
+---
+
+# JavaScript: creating and translating a kata
+
+This article is intended for kata authors and translators who would like to create new content in JavaScript. It attempts to explain how to create and organize things in a way conforming to [authoring guidelines](/authoring/guidelines/), shows the most common pitfalls and how to avoid them.
+
+This article is not a standalone tutorial on creating kata or translations. It's meant to be a complementary, JavaScript-specific part of a more general set of HOWTOs and guidelines related to [content authoring](/authoring/). If you are going to create a JavaScript translation, or a new JavaScript kata from scratch, please make yourself familiar with the aforementioned documents related to authoring in general first. 
+
+
+## General info
+
+Any technical information related to the JavaScript setup on Codewars can be found on the [JavaScript reference](/languages/javascript/) page (language versions, available libraries, and setup of the code runner).
+
+## Description
+
+JavaScript code blocks can be inserted with JavaScript-specific part in [sequential code blocks](/references/markdown/extensions/#sequential-code-blocks):
+
+~~~
+```javascript
+
+...your code here...
+
+```
+~~~
+
+JavaScript-specific paragraphs can be inserted with [language conditional rendering](/references/markdown/extensions/#conditional-rendering):
+
+```
+~~~if:javascript
+
+...text visible only for JavaScript description...
+
+~~~
+
+~~~if-not:javascript
+
+...text not visible in JavaScript description...
+
+~~~
+```
+
+## Tasks and Requirements
+
+Some concepts don't always translate well to or from JavaScript. Because of this, some constructs should be avoided and some translations just shouldn't be done.
+- Avoid returning different data types depending on the situation (_"Return the result, or the string 'Error' if no result can be found..."_). JavaScript is dynamically typed, which is not the case for some other languages. Returning `null` might be appropriate in some situations, but throwing an error might be better in others.
+
+Some kata just should not be translated into JavaScript because it can be difficult to keep their initial idea:
+- Some kata might be meant for another language specifically. For example, a kata about learning to use Python's `itertools.takewhile` function to do something should not be translated to JavaScript.
```

Comment:
I do not like the wording here, but it is a good point which could be turned into a general guideline: do not translate kata which are meant to be specific to some particular language, runtime, or a library, even if equivalents in other languages are available. For example, translating a kata like _"Learning Python: Classes"_ into JS might be not the best idea, even though both languages have classes.

Basically I would either reword this point (I know what you mean, and it can be just a language thing, but the mention of `itertools.takewhile` seems to be oddly specific to me and pops out somewhat suddenly here), or turn it into a general guideline and put it in [translating a kata](https://docs.codewars.com/authoring/guidelines/translation/) guidelines, for example in [this section](https://docs.codewars.com/authoring/guidelines/translation/#conformity-to-overall-kata-idea).